sql/core/src/test/scala/org/apache/spark/sql/JoinSuite.scala

```diff
@@ -895,4 +895,18 @@ class JoinSuite extends QueryTest with SharedSQLContext {
       checkAnswer(res, Row(0, 0, 0))
     }
   }
+
+  test("SPARK-26352: join reordering should not change the order of columns") {
+    withTable("tab1", "tab2", "tab3") {
```

Comment:
can we reproduce the bug with temp view?

Reply:
That's more prone to be subject to `ConvertToLocalRelation`, i.e. you can't copy-and-paste the test case into a Spark shell and expect the same behavior. Would you still prefer to use temp view instead?